src/open_inwoner/cms/extensions/cms_menus.py

```diff
@@ -51,13 +51,18 @@ def modify(self, request, nodes, namespace, root_id, post_cut, breadcrumb):
                 node.common = ext
 
                 # modify menu check for page visibility
-                if ext.requires_auth and not request.user.is_authenticated:
-                    nodes.remove(node)
-                    continue
-                elif ext.requires_auth_bsn_or_kvk and request.user.login_type not in [
-                    LoginTypeChoices.digid,
-                    LoginTypeChoices.eherkenning,
-                ]:
+                auth_required_regular_user = (
+                    ext.requires_auth and not request.user.is_authenticated
+                )
+                auth_required_digid_eherkenning_user = (
+                    ext.requires_auth_bsn_or_kvk
+                    and request.user.login_type
+                    not in [
+                        LoginTypeChoices.digid,
+                        LoginTypeChoices.eherkenning,
+                    ]
+                )
+                if auth_required_regular_user or auth_required_digid_eherkenning_user:
```

Comment:
Again I prefer the original. By way of contrast: if you store `not request.user.is_authenticated` in a variable `user_is_not_authenticated`, you preserve all information from the original. The two have the same meaning, and anyone who reads `user_is_not_authenticated` on its own knows what it means. 

That's not the case here: with `auth_required_regular_user` you loose information. If I read the variable on its own I don't know what it means. In particular, the variable name doesn't convey that the user is not authenticated. In order to know this, I have to go back up in the code to check. Then what's the point of storing this condition in a variable to begin with? The ruff version doesn't do this, but it has the drawback (IMHO) of a convoluted conjunction/disjunction...

Reply:
"using it in the next statement should be sufficient context": yes, but the variable doesn't add any value because it's not used anywhere else. It's only purpose is to help appease the linter even though the original was just fine, and this shows (in my view). However, this is really a minor issue. We can go along with this and re-evaluate if cases like this come up again.